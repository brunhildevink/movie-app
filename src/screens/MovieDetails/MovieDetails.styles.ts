import styled from 'styled-components'

const EpisodeInformation = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 40px;
  border-bottom: 1px solid rgba(151, 151, 151, 0.2);
`

const Image = styled.div<{ img: string }>`
  background: ${({ img }) => `url(${img})`} center center no-repeat;
  background-size: cover;
  height: 100%;
  width: 100%;
`

const TitleInformation = styled.div`
  padding: 40px;
`

const Wrapper = styled.div`
  height: 100%;
  display: grid;
  grid-template-rows: 60% 15% 25%;
  grid-template-columns: 1fr;
`

export { EpisodeInformation, Image, TitleInformation, Wrapper }
