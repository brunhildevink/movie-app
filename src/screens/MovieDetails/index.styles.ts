import styled from 'styled-components'
import { breakpoints, colors, typography } from '../../styles'

const EpisodeInformation = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 40px;
  border-bottom: 1px solid rgba(151, 151, 151, 0.2);
`

const Image = styled.div<{ img: string }>`
  background: ${({ img }) => `url(${img})`} center center no-repeat;
  background-size: cover;
  width: 100%;
  height: 200px;
  grid-area: image;

  @media ${breakpoints.lg} {
    grid-area: unset;
    height: 100%;
  }
`

const NotFound = styled.div`
  background: ${colors.white};
  height: 100%;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;

  svg {
    opacity: 0.5;
  }
`

const Rating = styled.div`
  display: flex;
  align-items: center;
  grid-area: rating;

  svg {
    margin-right: 16px;
  }

  span {
    font-weight: ${typography.fontWeightBold};
  }
`

const TitleInformation = styled.div`
  padding: 40px;
  overflow-y: auto;
  grid-area: title;

  p {
    font-size: ${typography.fontSizeTextLarge}px;
    line-height: ${typography.lineHeightTextLarge};
  }
`

const Wrapper = styled.div`
  height: 100%;
  display: grid;
  grid-template-columns: 1fr;
  grid-template-areas:
    'rating'
    'title'
    'image';

  @media ${breakpoints.lg} {
    grid-template-areas:
      'image'
      'rating'
      'title';

    grid-template-rows: 60% 15% 25%;
  }
`

export { EpisodeInformation, Image, NotFound, Rating, TitleInformation, Wrapper }
