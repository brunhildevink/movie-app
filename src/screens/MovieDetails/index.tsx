import React from 'react'
import { EpisodeInformation, Image, Rating, TitleInformation, Wrapper } from './MovieDetails.styles'
import image from '../../assets/images/1.1.jpg'
import { Heading, Icon, Text } from '../../components'
import { colors } from '../../styles'

interface Props {
  date: string
  description: string
  episode: number
  rating: string
  title: string
}

const MovieDetails: React.FC<Props> = ({ date, description, episode, rating, title }) => (
  <Wrapper>
    <Image img={image} />
    <EpisodeInformation>
      <Heading.HeadingFour>
        Episode {episode} â€” {date}
      </Heading.HeadingFour>
      <Rating>
        <Icon.Star display="inline" fill={colors.yellow} height={24} title="rating" width={24} />
        <Heading.HeadingFour>
          <span>{rating}</span>/10
        </Heading.HeadingFour>
      </Rating>
    </EpisodeInformation>
    <TitleInformation>
      <Heading.HeadingThree>{title}</Heading.HeadingThree>
      <Text.Regular>{description}</Text.Regular>
    </TitleInformation>
  </Wrapper>
)

export default MovieDetails
