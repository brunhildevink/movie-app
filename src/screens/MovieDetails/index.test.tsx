import React from 'react'
import renderer from 'react-test-renderer'
import MovieDetails from '.'

const mock = {
  Title: 'Chapter 1: The Mandalorian',
  Year: '2019',
  Rated: 'TV-PG',
  Released: '12 Nov 2019',
  Season: '1',
  Episode: '1',
  Runtime: '39 min',
  Genre: 'Action, Adventure, Fantasy',
  Director: 'Dave Filoni',
  Writer: 'Jon Favreau, George Lucas',
  Actors: 'Pedro Pascal, Carl Weathers, Werner Herzog',
  Plot: 'A Mandalorian bounty hunter tracks a target for a well-paying, mysterious client.',
  Language: 'English',
  Country: 'United States',
  Awards: 'N/A',
  Poster:
    'https://m.media-amazon.com/images/M/MV5BNTViYzhmZTAtY2MzYi00ZTk1LTg5OGItYzc4MjBlYzlkNzU0XkEyXkFqcGdeQXVyMTkxNjUyNQ@@._V1_SX300.jpg',
  Ratings: [
    {
      Source: 'Internet Movie Database',
      Value: '8.6/10',
    },
  ],
  Metascore: 'N/A',
  imdbRating: '8.6',
  imdbVotes: '30126',
  imdbID: 'tt9095424',
  seriesID: 'tt8111088',
  Type: 'episode',
  Response: 'True',
}

test('MovieDetails renders correctly', () => {
  const tree = renderer.create(<MovieDetails data={mock} />).toJSON()
  expect(tree).toMatchSnapshot()
})
