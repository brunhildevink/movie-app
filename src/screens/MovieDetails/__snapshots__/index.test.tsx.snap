// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails renders correctly 1`] = `<div />`;
